samza-core/src/main/java/org/apache/samza/runtime/ContainerLaunchUtil.java

```diff
@@ -127,6 +139,8 @@ public void afterFailure(Throwable t) {
       heartbeatMonitor.stop();
     }
 
+    coordinatorStreamStore.close();
```

Comment:
Do you think it's beneficial put this in a `finally` block and the code between here and the `coordinatorStreamStore.init()` in a `try` to ensure the store gets closed?

Reply:
Done.